enhancements/sig-architecture/14-addon-cluster-proxy/README.md

```diff
@@ -0,0 +1,536 @@
+      name: apiserver-network-proxy
+  addonMeta:
+    displayName: apiserver-network-proxy
+    description: "https://github.com/kubernetes-sigs/apiserver-network-proxy"
+  addonConfiguration:
+    # Importing templates for creating real agent instances.
+    crd: managedproxyconfigurations.proxy.addon.open-cluster-management.io
+    name: default
+```
+
+##### CredentialProjection
+
+`CredentialProjection` is a cluster-scoped api for automatically applying
+service-account tokens w/ expected rbac permissions from each of the managed 
+clusters that installed the addon. If it works successfully, a secret w/ the
+same name as the projection will be created in the managed cluster's 
+namespace. Note that the `CredentialProjection` is created by 
+
+```yaml
+apiVersion: proxy.open-cluster-management.io/v1alpha1
+kind: CredentialProjection
+metadata:
+  name: default # The name of the projected service-account-token
+  ownerReferences:
+    - apiVersion: proxy.open-cluster-management.io/v1alpha1
+      kind: ManagedClusterAddon
+      name: apiserver-network-proxy
+      blockOwnerDeletion: true
+spec:
+  type: [X509Certificate|ServiceAccountToken]
+  role:
+    - apiGroups:
+        - ""
+      resources:
+        - events
+      verbs:
+        - "*"
+  clusterRole:
+    - apiGroups:
+        - ""
+      resources:
+        - events
+      verbs:
+        - "*"
+status:
+  lastObservedGeneration: <...>
+  appliedClusters:
+    - name: <cluster #1>
+      lastUpdatedTimestamp: <...>
+    - name: <cluster #2>
+      lastUpdatedTimestamp: <...>
+```
+  
+#### Proxy Security
+
+The following figure shows the three sets of mTLS certificates/keys
+involved in the proxy framework:
+
+1. The mTLS key-pair for establishing the tunnel.
+2. The mTLS key-pair for initiating the proxy service.
+3. The mTLS key-pair for accessing the endpoint k8s apiserver, 
+   alternatively it can be a service-account token.
+
+```
+
+                        hub   |  public   |  spoke
+                      network | internet  | network
+client ======> server ========|===========|=========>  agent 
+-------------------------- Proxy Tunnel ----------------------
+L4                 <----------- mTLS(1) ---------------       T0: Setting up the tunnel
+L5                 <------ grpc proxy protocol --------       T0: Setting up the tunnel
+------------------- Proxied Multiplexed Traffic --------------
+client -mTLS(2)->                                             T1: Connects proxy server   
+       --------->                                             T2: Requests a proxied stream
+       --------------------- mTLS(3) ----------------->       T3: Unwinds kube-apiserver mTLS
+       ----------------- k8s api requests ------------>       T4: Sends request payload 
+```
+
+##### Server <-> Proxy Authentication
+
+The tunnel from agent to server is secured by mTLS technique, the same way
+we're doing for securing kubernetes' apiserver. So it's almost safe to expose
```

Comment:
Here is a little concern, if we expose anp proxy-server directly, in the future we want to replace another proxy approach say remotedialer, it would be painful since the tunnel layer is not interchangeable.

Reply:
can you share me a link to the alternative proxy approach you mentioned? i agree the proxy framework doesnt have to be coupled with apiserver-network-proxy tho, we can leave a bit forward-compatibility for the alternatives.

Reply:
Sure, we have considered adding https://github.com/rancher/remotedialer as an alternative approach in the future.